### ISSUE TITLE
Support Canonical Import Path

### ISSUE THREAD

**Issue by garukun** [go-lang-plugin-org/go-lang-idea-plugin#2098]
Just curious, is there any plans to support [Canonical Import Path](http://tip.golang.org/doc/go1.4#canonicalimports) since 1.4?

**Comment by zolotov** [comment]
What kind of support are expecting from IDE?

**Comment by dlsniper** [comment]
In the following example:

```go
package pdf // import "rsc.io/pdf"
``

I would expect to have an error if I import `github.com/rsc/pdf` instead of `rsc.io/pdf`. The quickfix should offer to change the import to `rsc.io/pdf`. Also I would expect the IDE doesn't offer `github.com/rsc/pdf` as a possible import candidate.
Does it make sense? Is it clear enough?

**Comment by zolotov** [comment]
No, it's not clear to me. Please provide an example project which wouldn't be compiled but IDEA won't find any error in it.

**Comment by dlsniper** [comment]
Hi, please see the code in this repository: https://github.com/dlsniper/canonical
Do not use `go get` but rather clone the repository in `$GOPATH/src/github.com/dlsniper/canonical` and then, in the IDE, open the `non_canonical.go` file and use the go get feature from there. As it's shown in `snapshot5.png`, go get fails and the IDE doesn't show any error about the import afterwards. However, if you open `canonical.go` everything is as intended.

**Comment by zolotov** [comment]
Still don't understand what IDE should do in this case and why errors from `go get` could affect its behavior. The code still compiles, right? And `go build` resolves `import "github.com/rsc/pdf"` just fine, doesn't it?

**Comment by dlsniper** [comment]
Sorry, I've left that part out.

```
cd $GOPATH/src/github.com/dlsniper/canonical
go build -a
non_canonical.go:3:8: code in directory /home/florin/golang/src/github.com/rsc/pdf expects import "rsc.io/pdf"
```

The IDE should flag the import as a bad import path and suggest changing it to the canonical import path `rsc.io/pdf`.

**Comment by dlsniper** [comment]
It would also be good to exclude it from the completion path.
As for showing errors in the IDE, this can be solved as an inspection + quickfix, we just need support for getting the canonical import path from it.
